aignostic/aignostic/router/api.py

```diff
@@ -1,23 +1,113 @@
-from pydantic import BaseModel
-from fastapi import APIRouter
+from pydantic import BaseModel, HttpUrl
+from fastapi import APIRouter, HTTPException
+import requests
+import aignostic.metrics.metrics as metrics_lib
+
 
 api = APIRouter()
 
 
+class DatasetRequest(BaseModel):
+    datasetURL: HttpUrl
+    modelURL: HttpUrl
+    metrics: list[str]
+
+
+@api.post("/evaluate")
+async def generate_metrics_from_info(request: DatasetRequest):
+    """
+    Frontend will post URLs, metrics, etc., as JSON to this endpoint.
+    This function validates, processes, and forwards the data to the controller.
+    """
+    # Extract data from the validated request
+    datasetURL = request.datasetURL
+    modelURL = request.modelURL
+    metrics = request.metrics
+
+    results = await process_data(datasetURL, modelURL, metrics)
+
+    return {"message": "Data successfully received", "results": results}
+
+
+# Sanity check endpoint
+# This for checking time of last deployment
 @api.get("/")
-def hello():
-    return {"message": "Hello Worlds!"}
+def info():
+    return {"message": "Pushed at 21/01/2025 07:32"}
+
+
+async def process_data(datasetURL: HttpUrl, modelURL: HttpUrl, metrics: list[str]):
+    """
+    Controller function. Takes data from the frontend, received at the endpoint and then:
```

Comment:
Tip: install a doc comment generator plugin to generate a doc comment template when you start typing one out (since sticking to the format manually is painful)

Reply:
This should have been a regular comment but GitHub didn't give me that option